v2/uuid/pgxcompat/uuid.go

```diff
@@ -0,0 +1,258 @@
+package pgxcompat
+
+import (
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgtype"
+
+	"github.com/SKF/go-utility/v2/uuid"
+)
+
+const uuidBinaryLength = 16
+const uuidStrShortLength = 32
+const uuidStrFullLength = 36
+
+var errUndefined = errors.New("cannot encode status undefined")
+var errBadStatus = errors.New("invalid status")
+
+type UUID struct {
+	UUID   uuid.UUID
+	Status pgtype.Status
+}
+
+type Getter interface {
+	Get() interface{}
+}
+
+func (u *UUID) Set(value interface{}) error { // nolint:gocyclo
+	if value == nil {
+		*u = UUID{Status: pgtype.Null}
+		return nil
+	}
+
+	if getterValue, ok := value.(Getter); ok {
+		value2 := getterValue.Get()
+		if value2 != getterValue {
+			return u.Set(value2)
+		}
+	}
+
+	switch value := value.(type) {
+	case uuid.UUID:
+		*u = UUID{UUID: value, Status: pgtype.Present}
+	case [16]byte:
+		uuidStr, _ := fromBinary(value[:]) // nolint:errcheck // we know input is 16 bytes
+		*u = UUID{UUID: uuid.UUID(uuidStr), Status: pgtype.Present}
+	case []byte:
+		if value == nil {
+			*u = UUID{Status: pgtype.Null}
+		} else if uuidStr, err := fromBinary(value[:]); err != nil {
+			return err
+		} else {
+			*u = UUID{UUID: uuid.UUID(uuidStr), Status: pgtype.Present}
+		}
+	case string:
+		if len(value) == uuidStrShortLength {
+			value = fmt.Sprintf("%s-%s-%s-%s-%s", value[0:8], value[8:12], value[12:16], value[16:20], value[20:32])
+		}
+
+		*u = UUID{UUID: uuid.UUID(value), Status: pgtype.Present}
+	case *string:
+		if value == nil {
```

Comment:
instead of `value == nil`, we can do a `case nil:` and return UUID  pgtype.Null there.